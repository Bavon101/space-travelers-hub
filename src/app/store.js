import { combineReducers, createStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import { missionsReducer } from '../components/missions/redux/missions';
import rocketReducer from '../components/rockets/redux/rockets';
import fetchRocketThunk from '../components/rockets/redux/rocketThunk';

const rootReducer = combineReducers({
  missionsReducer,
  rocketReducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk));
store.dispatch(fetchRocketThunk());

export default store;
