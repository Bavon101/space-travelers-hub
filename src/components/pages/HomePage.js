import React from 'react';
import { useSelector } from 'react-redux';
import RocketCard from '../rockets/rockets';

const HomePage = () => {
  const data = useSelector((state) => state.rocketReducer.rockets);
  if (data.length === 0) {
    return (
      <div>
        <ul>
          <li>
            <h1>No rockets available</h1>
          </li>
        </ul>
      </div>
    );
  }
  return (
    <ul>
      {data.map(({
        id, rocketName, description, flickrImages, reserved,
      }) => (
        <RocketCard
          key={id}
          rocketName={rocketName}
          description={description}
          flickrImages={flickrImages}
          reserved={reserved}
          id={id}
        />
      ))}
    </ul>
  );
};

export default HomePage;
