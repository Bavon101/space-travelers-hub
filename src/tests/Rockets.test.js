import React from 'react';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '../app/store';
import HomePage from '../components/pages/HomePage';

const RocketsProvider = () => (
    <Provider store={store}>
        <HomePage />
    </Provider>
);


describe ('Check Homepage', () => {
    it('renders correctly', () => {
       const component = renderer.create(<RocketsProvider />).toJSON();
         expect(component).toMatchSnapshot();
    });
});

describe ('No Rocket Available', () => {
    it('renders correctly', () => {
        render(<RocketsProvider />);
        expect(screen.getByText('No rockets available')).toBeInTheDocument();
    });
});

describe('Rockets Component', () => {
    it('renders rockets', async () => {
      render(<RocketsProvider />);
      const allRockets = await screen.findAllByText('Reserve Rocket');
      expect(allRockets).toHaveLength(4);
    }); 


test ('Reserve a rocket and cancel reservation', async () => {
    render(<RocketsProvider />);
    const reserveButton = await screen.findAllByText('Reserve Rocket');
    const badges = await screen.findAllByText('Reserved');
    expect(reserveButton[0].textContent).toBe('Reserve Rocket');
    fireEvent.click(reserveButton[0]);
    expect(badges[0].textContent).toBe('Reserved');
    expect(reserveButton[0].textContent).toBe('Cancel Reservation');
    });
});