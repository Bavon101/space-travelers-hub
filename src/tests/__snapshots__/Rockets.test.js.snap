// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Homepage renders correctly 1`] = `
<div>
  <ul>
    <li>
      <h1>
        No rockets available
      </h1>
    </li>
  </ul>
</div>
`;
